examples/shared/src/main/scala/show.scala

```diff
@@ -20,26 +20,30 @@ trait GenericShow[Out] {
 
   /** creates a new [[Show]] instance by labelling and joining (with `mkString`) the result of
     *  showing each parameter, and prefixing it with the class name */
-  def combine[T](ctx: CaseClass[Typeclass, T]): Show[Out, T] = new Show[Out, T] {
-    def show(value: T) =
-      if (ctx.isValueClass) {
-        val param = ctx.parameters.head
-        param.typeclass.show(param.dereference(value))
-      } else {
-        val paramStrings = ctx.parameters.map { param =>
-          s"${param.label}=${param.typeclass.show(param.dereference(value))}"
+  def combine[T](ctx: CaseClass[Typeclass, T]): Show[Out, T] = (value: T) => {
```

Comment:
My personal preference would be to write this as literally, `= { value =>`, not `= value => {`. (Removing the type ascription, too.)